website/docs/about_code_generation.mdx

```diff
@@ -55,13 +79,133 @@ This includes but is not limited to:
 - some Riverpod features will be available only with code generation.
 
 At the same time, many applications already use code generation with packages such
-as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).  
+as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## Provider type
+
+### <span style={RED_STYLE}>[Sync/Async]</span><span style={BLUE_STYLE}>[Stateless/Stateful]</span>Provider
```

Comment:
I'm still a bit on the fence about `Stateful/Stateless`.  
Technically all providers return a state. So it's a bit confusing.

Maybe `Functional/Class-based` or `Unmodifiable/Modifiable`. Something among those lines.

Reply:
`Function/Class-based` is great.
`Unmodifiable/Modifiable` is a bit confusing as users might think state won't change, as `Unmodifiable` is well-known for that when used with collections